etl/data_enhancement/enhance_data.py

```diff
@@ -1,20 +1,51 @@
-from importlib import import_module
+from .enhancement_tags import enhancement_tags as e_tags
+from .enhancement_location import enhancement_location as e_location
+from .enhancement_location.lat_lon_enhancer import add_lat_lon
 
 
-def run(data, domain_name):
-    load_enhancement_by_file(data, domain_name, 'default')
-    load_enhancement_by_file(data, domain_name)
+class Enhancer:
+    """ Class handling the enhancement steps for crawled data based on its origin """
 
-    return data
+    def __init__(self, data, domain_name):
+        self.__data = data
+        self.__domain_name = domain_name
 
+        # link function containing domain specific enhancement to said domain here
+        # 'domain' : self.__enhance_domain_function_name
+        # make sure to enter 'self.__enhance_domain_function_name' and not
+        # 'self.__enhance_domain_function_name()' as brackets would make this
+        # a function call instead of a reference to the function
+        self.__function_map = {
+            'ehrenamt_hessen': self.__enhance_ehrenamt_hessen,
+            'weltwaerts': self.__enhance_weltwaerts,
+            'gutetat_berlin': self.__enhance_gute_tat,
+            'gutetat_hamburg': self.__enhance_gute_tat,
+            'gutetat_munich': self.__enhance_gute_tat,
+        }
 
-def load_enhancement_by_file(data, domain_name, file_name=None):
-    try:
-        if not file_name:
-            file_name = domain_name
+    def run(self):
+        """ run general enhancements and load domainspecific enhancement. """
+        print(f"Run general enhancement enhancement for {self.__domain_name}")
+        e_location.add_map_address(self.__data)
+        self.__run_for_domain(self.__domain_name)
+        return self.__data
 
-        enhancement = import_module(f'data_enhancement.domain.{file_name}')
-        enhancement.run(data)
-        print(f'Enhanced data from [{domain_name}] with {file_name}.py')
-    except Exception as err:
-        print(err)
+    def __run_for_domain(self, domain):
+        """ run domainspecific enhancement. """
+        print(f"Run domain specific enhancement enhancement for {domain}")
+        if domain in self.__function_map:
+            self.__function_map[domain]()
+        else:
+            print(f"Error [enhance_data.py]: No function set for '{domain}'")
+
+    def __enhance_ehrenamt_hessen(self):
+        """ domain specific enhancement for ehrenamtsuche hessen """
+        e_tags.run(self.__data, self.__domain_name)
+
+    def __enhance_weltwaerts(self):
+        """ domain specific enhancement for weltwaerst """
+        add_lat_lon(self.__data)
```

Comment:
add_lat_lon shall be called for every website.

For ehrenamt_hessen 'geo_location = None' is not implemented yet. Maybe this can be done with this PR.
```
'geo_location': {
                'lat': lat,
                'lon': lon,
            } if lat and lon else None, # If longitude and latitude are None, geo_location is set to None
```

Reply:
I will implement the change for this PR

Reply:
Added lat_lon enhancement to default enhancement, added 'geo_location = None'